File: scalang/src/main/scala/com/kapilkoju/autopos/config/ThymeleafConfiguration.scala
package com.kapilkoju.autopos.config

import org.apache.commons.lang3.CharEncoding
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.context.annotation._
import org.thymeleaf.templateresolver.ClassLoaderTemplateResolver

@Configuration class ThymeleafConfiguration {
  @SuppressWarnings(Array("unused")) final private val log: Logger = LoggerFactory.getLogger(classOf[ThymeleafConfiguration])

  @Bean
  @Description("Thymeleaf template resolver serving HTML 5 emails") def emailTemplateResolver: ClassLoaderTemplateResolver = {
    val emailTemplateResolver: ClassLoaderTemplateResolver = new ClassLoaderTemplateResolver
    emailTemplateResolver.setPrefix("mails/")
    emailTemplateResolver.setSuffix(".html")
    emailTemplateResolver.setTemplateMode("HTML5")
    emailTemplateResolver.setCharacterEncoding(CharEncoding.UTF_8)
    emailTemplateResolver.setOrder(1)
    return emailTemplateResolver
  }
}

File: src/main/kotlin/com/kapilkoju/autopos/config/ThymeleafConfiguration.kt
package com.kapilkoju.autopos.config

import org.apache.commons.lang3.CharEncoding
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Description
import org.thymeleaf.templateresolver.ClassLoaderTemplateResolver

@Configuration
class ThymeleafConfiguration {

    @Bean
    @Description("Thymeleaf template resolver serving HTML 5 emails")
    fun emailTemplateResolver(): ClassLoaderTemplateResolver {
        val emailTemplateResolver = ClassLoaderTemplateResolver()
        emailTemplateResolver.prefix = "mails/"
        emailTemplateResolver.suffix = ".html"
        emailTemplateResolver.templateMode = "HTML5"
        emailTemplateResolver.characterEncoding = CharEncoding.UTF_8
        emailTemplateResolver.order = 1
        return emailTemplateResolver
    }
}